src/main/scala/coreplex/BaseCoreplex.scala

```diff
@@ -35,58 +35,145 @@ trait HasCoreplexParameters {
 }
 
-case class CoreplexConfig(
-    nTiles: Int,
-    nExtInterrupts: Int,
-    nSlaves: Int,
-    nMemChannels: Int,
-    hasSupervisor: Boolean)
-{
-  val nInterruptPriorities = if (nExtInterrupts <= 1) 0 else (nExtInterrupts min 7)
-  val plicKey = PLICConfig(nTiles, hasSupervisor, nExtInterrupts, nInterruptPriorities)
+case class CoreplexParameters(implicit val p: Parameters) extends HasCoreplexParameters
+
+abstract class BareCoreplex(implicit val p: Parameters) extends LazyModule
+abstract class BareCoreplexBundle[+L <: BareCoreplex](_outer: L) extends Bundle {
+  val outer = _outer
+}
+abstract class BareCoreplexModule[+L <: BareCoreplex, +B <: BareCoreplexBundle[L]](_outer: L, _io: () => B) extends LazyModuleImp(_outer) {
+  val outer = _outer
+  val io = _io ()
 }
 
-abstract class BaseCoreplex(c: CoreplexConfig)(implicit p: Parameters) extends LazyModule
+trait CoreplexNetwork extends HasCoreplexParameters {
+  this: BareCoreplex =>
 
-abstract class BaseCoreplexBundle(val c: CoreplexConfig)(implicit val p: Parameters) extends Bundle with HasCoreplexParameters {
-  val master = new Bundle {
-    val mem = Vec(c.nMemChannels, new ClientUncachedTileLinkIO()(outerMemParams))
-    val mmio = new ClientUncachedTileLinkIO()(outerMMIOParams)
-  }
-  val slave = Vec(c.nSlaves, new ClientUncachedTileLinkIO()(innerParams)).flip
-  val interrupts = Vec(c.nExtInterrupts, Bool()).asInput
-  val debug = new DebugBusIO()(p).flip
-  val clint = Vec(c.nTiles, new CoreplexLocalInterrupts).asInput
-  val resetVector = UInt(INPUT, p(XLen))
-  val success = Bool(OUTPUT) // used for testing
+  val l1tol2 = LazyModule(new TLXbar)
+  val l1tol2_beatBytes = p(TLKey("L2toMMIO")).dataBitsPerBeat/8
+  val l1tol2_lineBytes = p(CacheBlockBytes)
+
+  val cbus = LazyModule(new TLXbar)
+  val cbus_beatBytes = p(XLen)/8
+  val cbus_lineBytes = l1tol2_lineBytes
 
-  override def cloneType = this.getClass.getConstructors.head.newInstance(c, p).asInstanceOf[this.type]
+  val mmio = TLOutputNode()
+  val mmioInt = IntInputNode()
+
+  cbus.node :=
+    TLAtomicAutomata(arithmetic = true)( // disable once TLB uses TL2 metadata
+    TLWidthWidget(l1tol2_beatBytes)(
+    TLBuffer()(
+    l1tol2.node)))
+
+  mmio :=
+    TLBuffer()(
+    TLWidthWidget(l1tol2_beatBytes)(
+    l1tol2.node))
 }
 
-abstract class BaseCoreplexModule[+L <: BaseCoreplex, +B <: BaseCoreplexBundle](
-    c: CoreplexConfig, l: L, b: => B)(implicit val p: Parameters) extends LazyModuleImp(l) with HasCoreplexParameters {
-  val outer: L = l
-  val io: B = b
+trait CoreplexNetworkBundle extends HasCoreplexParameters {
+  this: {
+    val outer: CoreplexNetwork
+  } =>
+
+  implicit val p = outer.p
+  val mmio = outer.mmio.bundleOut
+  val interrupts = outer.mmioInt.bundleIn
+}
+
+trait CoreplexNetworkModule extends HasCoreplexParameters {
+    this: BareCoreplexModule[BareCoreplex, BareCoreplexBundle[BareCoreplex]] =>
+  implicit val p = outer.p
+}
+
+trait CoreplexRISCV {
```

Comment:
We have BareCoreplex and BaseCoreplex, the classes. Then the traits are CoreplexNetwork and CoreplexRISCV. Calling one just Coreplex makes it ambiguous...